src/vm/appdomain.cpp

```diff
@@ -12583,6 +12583,8 @@ AppDomain::RaiseAssemblyResolveEvent(
             if ( pSpec->IsIntrospectionOnly() 
 #ifdef FEATURE_FUSION
                     || pSpec->GetParentLoadContext() == LOADCTX_TYPE_UNKNOWN
+#elif FEATURE_CORECLR
+                    || pSpec->GetBindingContextFromParentAssembly(this) != GetTPABinderContext()
 #endif
```

Comment:
Ok will do. But most of the time they will still get NULL as for most of Load parentassembly is NULL. In all of coreclr tests I only got non NULL parentassembly when Type.GetType() is called.